src/pydata_sphinx_theme/__init__.py

```diff
@@ -902,9 +926,15 @@ def _overwrite_pygments_css(app, exception=None):
         # see if user specified a light/dark pygments theme, if not, use the
         # one we set in theme.conf
         style_key = f"pygment_{light_or_dark}_style"
-        theme_name = app.config.html_theme_options.get(
-            style_key, app.builder.globalcontext.get(f"theme_{style_key}")
-        )
+
+        # globalcontext sometimes doesn't exist so this ensures we do not error
+        if hasattr(app.builder, "globalcontext"):
+            theme_name = app.config.html_theme_options.get(
+                style_key, app.builder.globalcontext.get(f"theme_{style_key}")
+            )
+        else:
+            theme_name = fallback
```

Comment:
I think this logic is not quite correct.  THis will use `fallback` when `globalcontext` is missing *even if `app.config.html_theme_options[style_key]` exists*. I think we want something like this instead:

```suggestion
        theme_name = app.config.html_theme_options.get(style_key, None)
        if theme_name is None and hasattr(app.builder, "globalcontext"):
            theme_name = app.builder.globalcontext.get(f"theme_{style_key}",
                                                       None)
        if theme_name is None:
            theme_name = fallback
```